### PR TITLE
[Merged by Bors] - Support the openvino async

### LINKED ISSUE #370: Support async mode in OpenVINO
The performance of OpenVINO in async mode is better than sync mode. Adlik should think about use this recommendation working mode.

### PR DESCRIPTION
**Pull request by wangzhenyu32**
Closes #370

### CODE REVIEW

File: adlik_serving/runtime/openvino/model/ir_model.cc [hunk 1]
```diff
@@ -269,11 +242,37 @@ IRModel::IRModel(const ModelConfig& config, const ModelId& model_id) : config(co
...
+  InferenceEngine::Core core;
+  try {
+    network = core.ReadNetwork(xmlFileName, binFileName);
+  } catch (const details::InferenceEngineException& e) {
+    INFO_LOG << "Cannot load the model";
+    return tensorflow::errors::Internal("Cannot load the model ", e.what());
+  }
+  // 3.set batch size
+  auto currentBatchSize = network.getBatchSize();
+  if (currentBatchSize != config.max_batch_size()) {
```
Comment (amadeus-zte) [inline adlik_serving/runtime/openvino/model/ir_model.cc:261]: Is it proper?

File: adlik_serving/runtime/openvino/model/ir_model.cc [hunk 2]
```diff
@@ -269,11 +242,36 @@ IRModel::IRModel(const ModelConfig& config, const ModelId& model_id) : config(co
...
+    network.setBatchSize(config.max_batch_size());
+  }
+  // 4. set instance
   for (const auto& group : config.instance_group()) {
-    std::unique_ptr<BatchProcessor> instance;
-    TF_RETURN_IF_ERROR(createInstance(group.name(), config, config.getModelPath(model_id), &instance));
-    add(std::move(instance));
+    ExecutableNetwork executableNetwork = core.LoadNetwork(network, "CPU");
```
Comment (shiyingjin) [inline adlik_serving/runtime/openvino/model/ir_model.cc:266]: Can ExecutableNetwork be shared by multiple infer_request?

File: adlik_serving/runtime/openvino/model/ir_model.cc (hunk 1 above)
Comment (shiyingjin) [reply adlik_serving/runtime/openvino/model/ir_model.cc:261]: Yes

**Review by shiyingjin — CHANGES_REQUESTED** [review]

**Review by shiyingjin — APPROVED** [review]

**Comment by amadeus-zte** [comment]
bors retry

**Pull request closed without merging**